Auth.py

```diff
@@ -299,4 +306,78 @@ def blacklist_token(self, token):
             self.blacklisted_tokens.append(token_list[i]["token"])
 
         conn.commit()
-        conn.close()
\ No newline at end of file
+        conn.close()
+        
+    def sign_up_test(self):
+        #Make sure the user is already deleted
+        conn = pymysql.connect(self.host, self.username, self.password, self.database, cursorclass=pymysql.cursors.DictCursor)
+        cursor = conn.cursor()
+
+        cursor.execute("DELETE FROM users WHERE (firstname = 'Test' AND lastname = 'Test' );")
+
+        conn.commit()
+        conn.close()
+            
+        error_messages = ["empty-first-name", "empty-last-name", "empty-email", 
+                              "email-already-taken", "invalid-email", 
+                              "password-too-short", "passwords-no-match"]
+        result = self.sign_up( firstname="Test", lastname="Test", email="test@gmail.com", password="longpassword", confirmpassword="longpassword")
+
+        if result in error_messages:
+            print("Something went wrong in the sign_up_test, the error was: " + result)
+        else:
+            print("Everything went fine in the sign_up_test")
+        #These functions run the individual tests
+        
+
+    def sign_in_test(self):
+        error_messages = ["user-no-exist", "incorrect-email-or-password"]
+
+        result = self.sign_in( email="test@gmail.com", password="longpassword")
```

Comment:
Updating email here (just because "Gmail" is sort of a trademark name):

```suggestion
        result = self.sign_in( "test@example.com", "longpassword")
```